github/actions_secrets.go

```diff
@@ -16,6 +18,31 @@ type PublicKey struct {
 	Key   *string `json:"key"`
 }
 
+func (p *PublicKey) UnmarshalJSON(data []byte) error {
+	var pk struct {
+		KeyID interface{} `json:"key_id,string"`
+		Key   *string     `json:"key"`
+	}
+
+	if err := json.Unmarshal(data, &pk); err != nil {
+		return err
+	}
+
+	p.Key = pk.Key
+
+	switch v := pk.KeyID.(type) {
+	case string:
+		p.KeyID = &v
+	case float64:
+		p.KeyID = new(string)
+		*p.KeyID = strconv.FormatFloat(v, 'f', -1, 64)
```

Comment:
Why not just get rid of line 37?

```suggestion
		p.KeyID = String(strconv.FormatFloat(v, 'f', -1, 64))
```

Reply:
That was pretty obvious. Thanks!